bar.go

```diff
@@ -268,34 +268,27 @@ func (b *Bar) SetPriority(priority int) {
 // if bar is already in complete state. If drop is true bar will be
 // removed as well.
 func (b *Bar) Abort(drop bool) {
-	select {
-	case b.operateState <- func(s *bState) {
-		if s.completed == true {
-			return
-		}
-		if drop {
-			b.container.dropBar(b)
-			b.cancel()
-			return
-		}
-		go func() {
-			var uncompleted int
-			b.container.traverseBars(func(bar *Bar) bool {
-				if b != bar && !bar.Completed() {
-					uncompleted++
-					return false
-				}
-				return true
-			})
-			if uncompleted == 0 {
-				b.container.refreshCh <- time.Now()
+	if b.Completed() {
+		return
+	}
+
+	if drop {
+		b.container.dropBar(b)
```

Comment:
This is less atomic, as at time of call on line 276, bar may become completed.

Reply:
That’s exactly the same as in the case where we defer the `dropBar` into a separate Goroutine AFAICS: the Goroutine can be scheduled much later, and the bar may complete in the meantime.

Reply:
I.e. this is less atomic than the current code in the repo, but I don’t think it’s less atomic than the #101 variant.

Reply:
in #101 variant it's ok to move `b.cancel()` call outside anonymous goroutine. This will make sure bar is canceled while in incomplete state.